extension/goext/util.go

```diff
@@ -18,4 +18,10 @@ package goext
 type IUtil interface {
 	NewUUID() string
 	GetTransaction(context Context) (ITransaction, bool)
+
+	// ResourceFromContextForType converts mapped representation to structure representation of the resource for given type
+	ResourceFromContextForType(context map[string]interface{}, rawResource interface{}) (Resource, error)
```

Comment:
I don't like context name here since this can be a part of context or even unrelated to any context. I'd prefer a name like ResourceFromMap which gets mappedRes map[string] and the type